src/Machine.js

```diff
@@ -297,6 +299,15 @@ class Machine {
     // read
     const res = this.fileCache.read(reader, includePath);
 
+    // Check if source with same hash value has already been included 
```

Comment:
Is there a way to show the place in the code of the previous and the current inclusions? The warning should be actionable.